postCompose/src/main/java/com/hamdan/forzenbook/post/compose/Post.kt

```diff
@@ -110,16 +117,32 @@ private fun ImagePostContent(
     state: BasePostViewModel.PostState,
     onToggleClicked: () -> Unit,
     onSendClicked: () -> Unit,
+    onGalleryClicked: () -> Unit,
 ) {
     StandardContent(
         state = state,
         onToggleClicked = onToggleClicked,
-        onSendClicked = onSendClicked
+        onSendClicked = onSendClicked,
+        selected = (state as BasePostViewModel.PostState.Content).content is BasePostViewModel.PostContent.Text
     ) { padding ->
         BackgroundColumn(
             modifier = Modifier
                 .padding(padding),
         ) {
+            Spacer(modifier = Modifier.height(ForzenbookTheme.dimens.grid.x2))
+            SubmitButton(
+                label = stringResource(id = R.string.post_gallery_button_text),
+                enabled = true
+            ) {
+                onGalleryClicked()
+            }
+            ((state as BasePostViewModel.PostState.Content).content as BasePostViewModel.PostContent.Image).filePath?.apply {
```

Comment:
These `as` can fail. Need more elegant casting. Make some nice extensions

Reply:
I thought the point was so that they do fail when used incorrectly. Like how you were saying that one day if were needing it to do something as a specific state its expected to be and it isn't that state, then we probably should crash. Or I may be remembering wrong.

Reply:
It should fail if you're wrapping it in a try and looking for it to fail in debug mode, etc. For production code you want to be resilient

Reply:
I see, I misunderstood on that day then.

Reply:
Ok I think I implemented them in a rather nice way, following a xOrNull pattern